clients/src/test/java/org/apache/kafka/common/message/MessageTest.java

```diff
@@ -749,6 +750,13 @@ public void testProduceResponseVersions() throws Exception {
         }
     }
 
+    @Test
+    public void defaultValueShouldBeWritable() {
+        for (short version = SimpleExampleMessageData.LOWEST_SUPPORTED_VERSION; version <= SimpleExampleMessageData.HIGHEST_SUPPORTED_VERSION; ++version) {
+            MessageUtil.toByteBuffer(new SimpleExampleMessageData(), version);
```

Comment:
This fails without the change in this PR?

Reply:
> This fails without the change in this PR?

yep. The generated code is shown below.
```java
            if (this.taggedUuid != Uuid.fromString("H3KKO4NTRPaCWtEmm3vW7A")) {
                throw new UnsupportedVersionException("Attempted to write a non-default taggedUuid at version " + _version);
            }
```